src/big_core_input.cpp

```diff
@@ -39,6 +39,9 @@ void BigCoreInput::openFile(const std::string& filename)
 		throw "File format is not BIG format!";
 	}
 
+	// this vector must be empty before loading any data!
+	this->dataType.clear();
+
```

Comment:
As method readData is appending values to vector, it needs to be first reseted when reusing already created instance of BigCoreInput.